src/Database/LSMTree/Internal/Run/Index/Compact.hs

```diff
@@ -424,26 +424,31 @@ import           Database.LSMTree.Internal.Serialise
 -- See [a semi-formal description of the compact index](#rep-descr) for more
 -- details about the representation.
 data CompactIndex = CompactIndex {
```

Comment:
Is the re-ordering technically necessary for serialsations/deserialisation? If not, I prefer the previous ordering because it puts the components in order of when they are used in the search algorithm, and it matches the LateX equations above

Reply:
No, it's not technically necessary, just seemed like it makes sense to have it match the file format. I can drop that commit.

Reply:
I'll just do the re-ordering for the chunks, as they are the data structure that gets serialised.